vim25/types/hardware_version.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright (c) 2024-2024 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"golang.org/x/mod/semver"
+)
+
+// hardwareVersions is a list of supported hardware versions from
+// https://kb.vmware.com/s/article/1003746.
+var hardwareVersions = []HardwareVersion{
```

Comment:
Reminds me, we haven't updated hardwareVersions used by govc, but we can take care of that later: https://github.com/vmware/govmomi/blob/9e1e574278d4d93cd5b5ed2f1d030bbe341866cc/govc/vm/create.go#L35-L44

Reply:
I can take a look at that.